atc/resource/fetcher.go

```diff
@@ -71,6 +73,9 @@ func (f *fetcher) Fetch(
 	defer ticker.Stop()
 
 	versionedSource, err := f.fetchWithLock(ctx, logger, source, imageFetchingDelegate.Stdout())
+	if containerLock != nil {
+		containerLock.Release()
+	}
```

Comment:
the lock would also need to be released after the call to `fetchWithLock` on line 86 below.